File: src/web/magmac/app/compile/rule/ExactRule.ts
export class ExactRule {
	public lex( input : String) : CompileResult<Node> {if(true){ return CompileResults.fromResult( new Ok<>( new MapNode( )));;}return CompileErrors.createStringError( "Slice '" + this.value + "' not present", input);;}
	public generate( node : Node) : CompileResult<String> {return CompileResults.fromResult( new Ok<>( this.value));;}
}

File: packages/compiler-java/src/main/java/magmac/app/compile/rule/ExactRule.java
package magmac.app.compile.rule;

import magmac.api.result.Ok;
import magmac.app.compile.error.CompileResult;
import magmac.app.compile.error.CompileResults;
import magmac.app.compile.error.type.CompileErrors;
import magmac.app.compile.node.MapNode;
import magmac.app.compile.node.Node;

public record ExactRule(String value) implements Rule {
    @Override
    public CompileResult<Node> lex(String input) {
        if (input.equals(this.value)) {
            return CompileResults.fromResult(new Ok<>(new MapNode()));
        }

        return CompileErrors.createStringError("Slice '" + this.value + "' not present", input);
    }

    @Override
    public CompileResult<String> generate(Node node) {
        return CompileResults.fromResult(new Ok<>(this.value));
    }
}
